src/Components/BranchesControl.jsx

```diff
@@ -0,0 +1,132 @@
+import React, {useContext, useEffect, useState} from 'react'
+import {useNavigate} from 'react-router-dom'
+import TextField from '@mui/material/TextField'
+import MenuItem from '@mui/material/MenuItem'
+import Paper from '@mui/material/Paper'
+import Typography from '@mui/material/Typography'
+import {makeStyles} from '@mui/styles'
+import {ColorModeContext} from '../Context/ColorMode'
+import debug from '../utils/debug'
+import {getBranches} from '../utils/GitHub'
+import useStore from '../store/useStore'
+
+
+/**
+ * @param {boolean} isDialogDisplayed
+ * @param {Function} setIsDialogDisplayed
+ * @return {object} React component
+ */
+export default function Branches() {
+  const classes = useStyles()
+  const navigate = useNavigate()
+  const colorMode = useContext(ColorModeContext)
+  const repository = useStore((state) => state.repository)
+  const [branchNames, setBranchNames] = useState([])
+  const [versionPaths, setVersionPaths] = useState([])
+  const modelPath = useStore((state) => state.modelPath)
+  const minVersions = 2
+
+  useEffect(() => {
+    if (!repository) {
+      debug().warn('IssuesControl#Issues: 1, no repo defined')
+      return
+    }
+    const fetchBranches = async () => {
+      try {
+        const branchesData = await getBranches(repository)
+        const versionPathsLocal = []
+        if (branchesData.data.length > 0) {
+          setBranchNames(branchesData.data)
+        }
+        branchesData.data.map((branch) => {
+          modelPath.branch = branch.name
+          const versionPath = `/share/v/gh/${modelPath.org}/${modelPath.repo}/${modelPath.branch}${modelPath.filepath}`
+          versionPathsLocal.push(versionPath)
+        })
+        setVersionPaths(versionPathsLocal)
+      } catch (e) {
+        debug().warn('failed to fetch issues', e)
+      }
+    }
+    if (branchNames.length === 0) {
+      fetchBranches()
+    }
+  }, [repository])
+
+
+  const handleSelect = (event) => {
+    const versionNumber = event.target.value
+    navigate({
+      pathname: versionPaths[versionNumber],
+    })
+  }
+
+  return (
+    <Paper elevation={0}
+      sx={{
+        backgroundColor: colorMode.isDay() ? '#E8E8E8' : '#4C4C4C',
+        marginTop: '14px',
+        opacity: .8,
+        display: ( modelPath.repo !== undefined && branchNames.length ) < minVersions ? 'none' : 'box',
+      }}
+    >
+      <TextField
+        className={classes.dropDown}
+        onChange={(e) => handleSelect(e)}
+        variant='outlined'
+        label='Versions'
+        select
+      >
+        {branchNames.length > 0 && branchNames.map((branchName, i) => {
+          return (
+            <MenuItem
+              key={i}
+              value={i}
```

Comment:
What is the `value` property used for? It doesn't appear to be part of the defined props for `<MenuItem>`.

https://mui.com/material-ui/api/menu-item/